upload-api/ucan-invocation.js

```diff
@@ -1,46 +1,145 @@
 import * as CAR from '@ucanto/transport/car'
+import * as CBOR from '@ucanto/transport/cbor'
 import * as UCAN from '@ipld/dag-ucan'
 import * as Link from 'multiformats/link'
 import { fromString as uint8arrayFromString } from 'uint8arrays/from-string'
 
 import {
+  BadBodyError,
+  BadContentTypeError,
   NoTokenError,
   ExpectedBasicStringError,
-  NoValidTokenError
+  NoValidTokenError,
+  NoInvocationFoundForGivenReceiptError,
+  NoCarFoundForGivenReceiptError
 } from './errors.js'
 
+export const CONTENT_TYPE = {
+  WORKFLOW: 'application/invocations+car',
+  RECEIPT: 'application/receipt+dag-cbor'
+}
+
 /**
- * @typedef {import('./types').UcanInvocation} UcanInvocation
- * @typedef {import('./types').InvocationsCar} InvocationsCar
+ * @typedef {import('./types').UcanLogCtx} UcanLogCtx
+ * @typedef {import('./types').WorkflowCtx} WorkflowCtx
+ * @typedef {import('./types').ReceiptBlockCtx} ReceiptBlockCtx
+ * @typedef {import('./types').Workflow} Workflow
+ * @typedef {import('./types').ReceiptBlock} ReceiptBlock
  */
 
 /**
- * Persist CAR with handled UCAN invocations by the router.
- *
- * @param {InvocationsCar} invocationsCar
- * @param {import('@web3-storage/upload-api').UcanBucket} ucanStore
+ * @param {import('aws-lambda').APIGatewayProxyEventV2} request
+ * @param {UcanLogCtx} ctx
  */
-export async function persistInvocationsCar(invocationsCar, ucanStore) {
-  await ucanStore.put(invocationsCar.cid, invocationsCar.bytes)
+export async function processUcanLogRequest (request, ctx) {
+  const token = getTokenFromRequest(request)
+  if (token !== ctx.basicAuth) {
+    throw new NoValidTokenError('invalid Authorization credentials provided')
+  }
+
+  if (!request.body) {
+    throw new BadBodyError('service requests are required to have body')
+  }
+  const bytes = Buffer.from(request.body, 'base64')
+
+  const contentType = request.headers['Content-Type'] || ''
+  if (contentType === CONTENT_TYPE.WORKFLOW) {
+    return await processWorkflow(bytes, ctx)
+  } else if (contentType === CONTENT_TYPE.RECEIPT) {
+    return await processTaskReceipt(bytes, ctx)
```

Comment:
This is an invocation receipt.

```suggestion
    return await processInvocationReceipt(bytes, ctx)
```